### DISCUSSION TITLE
Network Switch Button Chains

### DISCUSSION THREAD

**Discussion by DorianDaSilva** [WalletConnect/walletconnect-monorepo#3560] [Developer Support] [answered]
**Description** :

I am importing & using <Web3NetworkSwitch/> in my 1st dApp build sólo but it is only allowing me to select 1 of 4 networks when connecting the wallet (Ethereum, BSC, Polygon & Avax)
_______________________

**Issue**:

It is not possible to connect on testnet to test the dApp or connect to any other other network besides the 4 mainnet ones available 
________________________

**Question**:

I looked at the "Web3modal" &  "wagmi" docs & tried to make it work but I am still unsure of how to implement the code to add more available networks to <Web3NetworkSwitch />

1- Should I create a new "connector" file to define the missing chains & then import it as a prop into my navbar component so <Web3NetworkSwitch /> can pull these additional networks for the user? 

2- Should I import the chains directly into the navbar component? 

3- Should I define it above the navbar function? 

Any help is appreciated!

Thank you!

**Comment by glitch-txs** [comment]
Hi, which wallets are you using to connect to the dapp? and are they mobile apps or browser extensions?

**Reply by DorianDaSilva** [reply, marked as answer]
Hi @glitch-txs 

thank you for taking the time to reply. I managed to figure it out earlier, it was quite simple. I guess I got confused.
